server/src/main/java/org/elasticsearch/cluster/AckedClusterStateUpdateTask.java

```diff
@@ -15,8 +15,9 @@
 import org.elasticsearch.core.TimeValue;
 
 /**
- * An extension interface to {@link ClusterStateUpdateTask} that allows to be notified when
- * all the nodes have acknowledged a cluster state update request
+ * An extension interface to {@link ClusterStateUpdateTask} that allows the caller to be notified after the master has
+ * accepted, processed, committed, and published the cluster state update AND after the rest of the nodes
```

Comment:
I would describe it as: process, publish, accept, commit, apply (and then ack). Although "process" is a little vague, maybe "computed" would be better.